sql/parser/eval_test.go

```diff
@@ -233,21 +233,23 @@ func TestEval(t *testing.T) {
 		// Cast expressions.
 		{`true::boolean`, `true`},
 		{`true::int`, `1`},
-		{`true::float`, `1`},
+		{`true::float`, `1.0`},
 		{`length(true::text)`, `4`},
 		{`false::boolean`, `false`},
 		{`false::int`, `0`},
-		{`false::float`, `0`},
+		{`false::float`, `0.0`},
 		{`length(false::text)`, `5`},
 		{`1::boolean`, `true`},
 		{`0::boolean`, `false`},
 		{`1::int`, `1`},
-		{`1::float`, `1`},
+		{`1::float`, `1.0`},
 		{`length(123::text)`, `3`},
 		{`1.1::boolean`, `true`},
 		{`0.0::boolean`, `false`},
 		{`1.1::int`, `1`},
 		{`1.1::float`, `1.1`},
+		{`999999.0`, `999999.0`},
+		{`1000000.0`, `1e+06`},
```

Comment:
These two cases now illustrate the turning point at which we deem whole floats 'large'